### ISSUE TITLE
Aligned samples lose their perfect alignment after retention time correction

### ISSUE THREAD

**Issue by etrh** [sneumann/xcms#313]
I have noticed that sometimes peaks that are perfectly aligned lose their alignment after retention time correction. I would like to know, if possible, which parameters lead to such a strange behavior. Interestingly, this only happens with some datasets and does not happen to other datasets at all.

Some examples can be seen below (in all of these cares upper panels show the raw data before alignment, and the lower panels show the samples after retention time correction).

![13 - rtime_alignment_mz-209 09684104327-209 097443240587_rt-990 315388068917-1097 28894924889](https://user-images.githubusercontent.com/29725715/45219956-e91e5b80-b2ac-11e8-9677-17dabd06aa50.png)
![13 - rtime_alignment_mz-315 175255365926-315 176158610004_rt-1115 3737483716-1222 35640281357](https://user-images.githubusercontent.com/29725715/45219957-e91e5b80-b2ac-11e8-9c99-5583c169a054.png)
![13 - rtime_alignment_mz-483 328326209349-483 32893732683_rt-1139 56191094135-1246 48265528021](https://user-images.githubusercontent.com/29725715/45219958-e91e5b80-b2ac-11e8-97f2-c4e6396f1923.png)
![13 - rtime_alignment_mz-610 388474186665-610 390300093374_rt-525 668718158703-632 647950418177](https://user-images.githubusercontent.com/29725715/45219959-e91e5b80-b2ac-11e8-938e-2afed22d9f34.png)
![13_-_rtime_alignment_evaluation_on_test_peak_before_after_alignment_2](https://user-images.githubusercontent.com/29725715/45219960-e9b6f200-b2ac-11e8-9a68-8d2f4757d53f.png)

This is the code that I used to get to these results:

```
## Correspondence: group peaks across samples
pdp <- PeakDensityParam(sampleGroups = xdata$sample_group, 
                        bw = 15, binSize = 0.05, 
                        minFraction = 0.5, minSamples = 8)

xdata <- groupChromPeaks(xdata, param = pdp)

## Retention time correction
pgp <- PeakGroupsParam(minFraction = 0.5, span = 0.55)

## Get the peak groups that would be used for alignment
xdata <- adjustRtime(xdata, param = pgp)


png("13 - rtime_alignment_evaluation_on_test_peak_before_after_alignment.png", 
    width = 15, height = 7, units = "in", res = 1200)
par(mfrow = c(2, 1))

## Plot the raw data
chr_raw <- chromatogram(raw_data, mz = mzr, rt = rtr)
plot(chr_raw, col = group_colors[chr_raw$sample_group])

## Extract the chromatogram from the adjusted object
chr_adj <- chromatogram(xdata, rt = rtr, mz = mzr)
plot(chr_adj, col = group_colors[chr_raw$sample_group]) 
dev.off()

```

**Comment by jotsetung** [comment]
Well yes, it can happen that nicely aligned peaks get mis-aligned after alignment. The methods don't align each single peak, but only the *hook peaks* (i.e. those peaks that are present in most samples). Alignment of spectra (and peaks) between these hook peaks is based on interpolation. It can happen that some previously aligned peaks get shifted. 
Probably you could play around with the `span` parameter to see if you can reduce this effect.

In my analyses I accept to get also some mis-aligned peaks as long as they are grouped together into the same feature by the following correspondence analysis.

**Comment by etrh** [comment]
@jotsetung Thank you. That explains it.

And do you have any idea why sometimes correspondence leads to cases such as the one depicted in the figure below? Note the number of red signals/peaks (several), and the number of red dots (just one).

![incorrect_peak_density_correspondence8200](https://user-images.githubusercontent.com/29725715/46147006-834a4180-c264-11e8-9360-7d5784df7d20.png)

This figures shows peak correspondence using three sets of different parameters.

**Comment by jotsetung** [comment]
Looks to me  that peak detection failed for some red lines (and also blue lines). That might explain why the correspondence did not define a feature for these. Depending on the value for `minFraction` and your `sampleGroups` it might just be that having e.g. only 4 peaks in the blue group was not enough to define the feature?

**Comment by etrh** [comment]
Just for clarification, are the dots supposed to represent the apex of a certain detected peak? I am just wondering why the dots do not fall on the signal in this case (they usually do with other dataset).

**Comment by jotsetung** [comment]
@etrh, yes, the dots represent the apex position reported by the peak detection algorithm. These can be the apex of a gaussian curve fitted to the peak data. If the peak is not symmetric, it can happen that the apex is not where one might expect it by looking at the data. So, I can understand the blue points to some extent, but the red point seems to be off. There was a problem with the color labeling in the `highlightChromPeak` function that I fixed at some point (could be that was not yet fixed in the version you used). 
Also, with the most recent `xcms` (that was released last week in Bioconductor 3.8) there is the possibility to use `type = "polygon"` in `highlightChromPeaks` that fills the detected peak area.
